spec/requests/portfolios_spec.rb

```diff
@@ -363,34 +355,34 @@
     context 'unshare' do
       include_context "sharing_objects"
       let(:unsharing_attributes) { {:group_uuids => group_uuids, :permissions => permissions} }
-      let(:dummy) { double("Insights::API::Common::RBAC::UnshareResource", :process => self) }
       it "portfolio" do
         with_modified_env :APP_NAME => app_name do
-          options = {:app_name      => app_name,
-                     :resource_ids  => [portfolio.id.to_s],
-                     :resource_name => 'portfolios',
-                     :permissions   => permissions,
-                     :group_uuids   => group_uuids}
-          expect(Insights::API::Common::RBAC::UnshareResource).to receive(:new).with(options).and_return(dummy)
-          post "#{api}/portfolios/#{portfolio.id}/unshare", :params => unsharing_attributes, :headers => default_headers
+          allow(rs_class).to receive(:call).with(RBACApiClient::GroupApi).and_yield(api_instance)
+          allow(Insights::API::Common::RBAC::Service).to receive(:paginate).with(api_instance, :list_groups, {}).and_return(groups)
+          ace1
+          ace2
+          ace3
```

Comment:
These `allow`s should probably be moved to a `before` as suggested above, and the `ace1, ace2, ace3`s should probably just be `let!`s in the shared context since you use them in both places you care about the context anyway.